### PR TITLE
"Hello world" project with rails-react gem

### PR DESCRIPTION
**Pull request by Cathella**
### Project requirements

- [ ] Create a new Rails app called 'hello-rails-react'.
- [ ] Initialize your project with Git.
- [ ] Set up your project with `webpacker` the react-rails gems as describe in the previous lesson's tutorial.
- [ ] Make sure that your project has a Postgres database set up.
- [ ] Create a table for storing your messages and create 5 different greetings.
- [ ] Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- [ ] Create a static view that will be the root of your app.
- [ ] Create the App component with react-router and render it in your static view.
- [ ] Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- [ ] Create a store, an action and a reducer that will connect to you API endpoint to get the random greeting.
- [ ] Display the random greeting in your Greeting component.

### CODE REVIEW

**Comment by rukundoeric** [comment]
Hi @Cathella, 

_While you made a great effort in this project, unfortunately, I cannot proceed to review your code._

## Invalid Code Review Request ⚠️

   - This project requires to use the correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md). which means you should have a PR from features into development not main. kindly fix it.



Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

>Invalid Code Review Request **does not count** into the code reviews limit.

File: app/javascript/components/Greeting.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchRandomGreeting } from "../redux/greetings";
+
+const Greeting = () => {
+  const dispatch = useDispatch();
```
Comment (rloterh) [inline app/javascript/components/Greeting.js:6]: - [ ] Your app is not working as intended. Kindly ensure that _rack-cors_ gem is installed and properly configured.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Nakitto, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- Descriptive Pull Request
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/components/Greeting.js (hunk 1 above)
Comment (Cathella) [reply app/javascript/components/Greeting.js:6]: Hi @rloterh, the app works perfectly on my side. And for the a react-rails project the `rack-cors` gem may or may not be needed. 
To run the app, 
- In the project root - run `bundle install`
- run `npm install`
- run migrations - `rails db:migrate`
- seed database data - `rails db:seed`
- run the server in the terminal at port 3000 by typing `rails s`
- In another terminal, in the project root - run `./bin/webpack-dev-server`
- load app at `localhost:3000`

Please check the video below 👇🏻 


https://user-images.githubusercontent.com/4247740/156060635-0c46e31b-111f-44b6-9b08-7e81c5bd7e06.mp4

Comment (rloterh) [reply app/javascript/components/Greeting.js:6]: ## Fellow Code Reviewer:

**Kindly approve this project!** This project is found to be complete upon further checking

Comment (rloterh) [reply app/javascript/components/Greeting.js:6]: @Cathella 
Yes, your project is found complete upon checking further. I missed a very important step during the review. Kindly note that this was not a deliberate attempt.

Cheers! and Happy coding! 😃

**Review by t-yanick — APPROVED** [review]
Hi @Cathella, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**